api/example.com/resource/gpu/v1alpha1/sharing.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2024 The Kubernetes Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import (
+	"fmt"
+)
+
+// These constants represent the different Sharing strategies.
+const (
+	TimeSlicingStrategy       GpuSharingStrategy = "TimeSlicing"
+	SpacePartitioningStrategy GpuSharingStrategy = "SpacePartitioning"
+)
+
+// These constants represent the different TimeSlicing configurations.
+const (
+	DefaultTimeSlice TimeSliceDuration = "Default"
+	ShortTimeSlice   TimeSliceDuration = "Short"
+	MediumTimeSlice  TimeSliceDuration = "Medium"
+	LongTimeSlice    TimeSliceDuration = "Long"
+)
+
+// Sharing provides methods to check if a given sharing strategy is selected
+// and grab its configuration.
+// +k8s:deepcopy-gen=false
+type Interface interface {
```

Comment:
Why is this an interface? As it stands now, I have to find the implementation to know what the semantic of each method is.

Remove it perhaps?

Reply:
This was relevant in the NVIDIA driver where this was pulled from. I don't remember the context though and would need to look back at that code base to understand why I did it this way. In any case, I agree, it is irrelevant here. Removing.

Reply:
Updated.